client/src/entrypoints/admin/core.js

```diff
@@ -305,15 +305,8 @@ $(() => {
     };
 
     // eslint-disable-next-line func-names
-    const getURLParam = function (name) {
-      const results = new RegExp('[\\?&]' + name + '=([^]*)').exec(
-        window.location.search,
-      );
-      if (results) {
-        return results[1];
-      }
-      return '';
-    };
+    const searchParams = new URLSearchParams(window.location.search);
+    searchParams.set('q', newQuery);
```

Comment:
I think there may be some confusion here. 

You should probably move the new URL... Thing up to line 270 - inside the function body of `search`

Also I do not see a need to set the query here at all - that's only needed if you intend to update the URL